l1b_lambda/level1b/handlers/level1b.py

```diff
@@ -156,8 +156,10 @@ def lambda_handler(event: Event, context: Context):
         raise Level1BException(msg) from err
 
     try:
-        for key in metadata.keys():
-            l1b_data[key] = metadata[key]
+        for key, val in metadata.items():
+            l1b_data[
+                key if isinstance(key, str) else key.encode()
+            ] = val if isinstance(val, str) else val.encode()
```

Comment:
It might be that `key.encode() if isinstance(key, bytes) else key` is better, not sure. Should make little difference, however.